### PR TITLE
fix version constraints for symfony 3

### PR DESCRIPTION
**Pull request by Ma27**
see schmittjoh/JMSDiExtraBundle#224
see schmittjoh/JMSDiExtraBundle#225

### CODE REVIEW

**Comment by rufinus** [comment]
:+1:

File: composer.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
         }
     ],
     "require": {
-        "symfony/framework-bundle": "2.*",
+        "php": "^5.5|~7.0",
+        "symfony/framework-bundle": "~2.8|^3.0",
```
Comment (Ener-Getick) [inline composer.json]: It isn't really clean to use this constraint as symfony 2.3 and 2.7 are still supported...

**Comment by Ener-Getick** [comment]
Fixed in #34

File: DependencyInjection/Compiler/PointcutMatchingPass.php [hunk 2]
```diff
@@ -42,18 +43,34 @@
  */
 class PointcutMatchingPass implements CompilerPassInterface
 {
+    /**
+     * @var array<PointcutInterface>
```
Comment (hoshsadiq) [inline DependencyInjection/Compiler/PointcutMatchingPass.php]: Should be `PointcutInterface[]`

File: DependencyInjection/Compiler/PointcutMatchingPass.php [hunk 3]
```diff
@@ -100,8 +119,14 @@ private function processDefinition(Definition $definition, $pointcuts, &$interce
             return;
         }
 
-        if ($definition->getFactoryService() || $definition->getFactoryClass()) {
-            return;
+        if (Kernel::VERSION_ID < 20600) {
```
Comment (hoshsadiq) [inline DependencyInjection/Compiler/PointcutMatchingPass.php]: Can combine these two if statements.

File: DependencyInjection/Compiler/PointcutMatchingPass.php [hunk 4]
```diff
@@ -100,8 +119,14 @@ private function processDefinition(Definition $definition, $pointcuts, &$interce
...
+            if ($definition->getFactory()) {
+                return;
+            }
+        } else {
```
Comment (hoshsadiq) [inline DependencyInjection/Compiler/PointcutMatchingPass.php]: Can be an `else if`

**Comment by Ma27** [comment]
closing because of #34

**Pull request closed without merging**